lamindb/_curate.py

```diff
@@ -737,7 +748,22 @@ def save_artifact(self, description: str | None = None, **kwargs) -> Artifact:
 
 
 class Curate:
-    """Annotation flow."""
+    """Curation flow.
+
+    Data annotation entails accurately labeling biological datasets with relevant metadata and functional information
+    to facilitate data integration, interpretation and analysis.
+
+    The curation flow has several steps:
+    1. Create an :class:`DataFrameCurator` (:meth:`~lamindb.Curate.from_df`), :class:`AnnDataCurator` (:meth:`~lamindb.Curate.from_anndata`),
+       :class:`MuDataCurator` (:meth:`~lamindb.Curate.from_mudata`) for the object type that you want to curate.
+       During the creation of the object, the passed categoricals will be validated to ensure that they are valid for curation.
```

Comment:
This sounds like categories will be validated. I'm somewhat struggling with this choice myself but @sunnyosun insisted it's the right choice: feature names are validated during instantiation.

But this means that we should call out "feature names" here. In theory, to arrive at invariance under feature renaming, we'd like to be able to pass uids at some point. I fear we'd need a flag to indicate what to validated against..., which is gonna be ugly.

Can we re-visit the idea of validating feature identifiers for metadata features during `.validate` just like we validate feature identifiers for the data during `.validate()`?

Reply:
Discussed with @sunnyosun today. Quoting her:
> columns need to be validated upon init because they are the keys in the categories, so it’d better to error already so people can fix passing categories

Reply:
Yes, she told me the same thing when we discussed it 4-5 months ago. I'm still not 100% convinced.

Reply:
If we're consistent, we should be able to pass `obs_columns=ln.Feature.name` or something like this. I think raising the error during `.validate()` would be more consistent with validating everything else but I might not see all arguments.

Reply:
adding obs_columns is straightforward: https://github.com/laminlabs/lamindb/pull/1790/commits/06637dd307968e3c8708c89390d01be9f7f94b71

Reply:
https://laminlabs.slack.com/archives/C04FPE8V01W/p1722956628159469